File: 0116_Populating Next Right Pointers in Each Node/PopulatingNextRightPointersinEachNode.cs
public class Solution {
    public Node Connect(Node root) {
        if(root == null) return null;
        if(root.left!=null)
        {
            root.left.next = root.right;
        }        

        if(root.next!=null && root.right!=null){
            root.right.next = root.next.left;
        }
        Connect(root.left);
        Connect(root.right);
        return root;
    }
}

File: 0116_Populating Next Right Pointers in Each Node/PopulatingNextRightPointersinEachNode.java
class Solution {
    public Node connect(Node root) {
        connect(root, null, true);
        return root;
    }
    
    private void connect(Node node, Node parent, boolean isLeft) {
        if(node == null) return;
        if(parent!=null) 
            node.next = isLeft ? parent.right : (parent.next == null ? null : parent.next.left);
        connect(node.left, node, true);
        connect(node.right, node, false);
    }
}